Python/Product/TestAdapter/TestContainerDiscovererProject.cs

```diff
@@ -153,11 +153,14 @@ class TestContainerDiscovererProject : ITestContainerDiscoverer, IDisposable {
             _testFilesUpdateWatcher.FileChangedEvent += OnProjectItemChanged;
             oldTestFilesUpdateWatcher?.Dispose();
 
-            var solution = (IVsSolution)_serviceProvider.GetService(typeof(SVsSolution));
-
-            // add all source files
-            foreach (var project in VsProjectExtensions.EnumerateLoadedProjects(solution)) {
-                LoadProject(project);
+            try {
+                // add all source files
+                var solution = (IVsSolution)_serviceProvider.GetService(typeof(SVsSolution));
+                foreach (var project in VsProjectExtensions.EnumerateLoadedProjects(solution)) {
+                    LoadProject(project);
+                }
+            } catch (Exception ex) when (!ex.IsCriticalException()) {
+                Trace.WriteLine("Exception : " + ex.Message);
```

Comment:
open project didn't have an issue ( probably because open folder as all source files under a single projectInfo, its more of a worst case)

added exception guard to make sure we set _setupComplete=true